apps/explorer/src/ui/Button.tsx

```diff
@@ -35,10 +36,20 @@ const buttonStyles = cva(
 
 export interface ButtonProps
     extends VariantProps<typeof buttonStyles>,
-        ButtonOrLinkProps {}
+        ButtonOrLinkProps {
+    loading?: boolean;
+}
 
-export function Button({ variant, size, ...props }: ButtonProps) {
+export function Button({
+    variant,
+    size,
+    loading,
+    children,
+    ...props
+}: ButtonProps) {
     return (
-        <ButtonOrLink className={buttonStyles({ variant, size })} {...props} />
+        <ButtonOrLink className={buttonStyles({ variant, size })} {...props}>
+            {loading ? <LoadingSpinner /> : children}
```

Comment:
It's probably better to actually just make loading do something like `text-transparent` and then display the loading spinner positioned absolutely in the button. That ensures the button is still correctly labelled, and avoids width changing.